### PR TITLE
Add OutputLocation to ResourceWrapper if bigquery job status is Done

### PR DESCRIPTION
**Pull request by pingsutw**
Signed-off-by: Kevin Su <pingsutw@apache.org>

# TL;DR
https://flyte-org.slack.com/archives/CNMKCU6FR/p1659118984659049

The problem is that BQ will [cache the query result](https://cloud.google.com/bigquery/docs/cached-results#:~:text=By%20default%2C%20BigQuery%20caches%20query,24%2Dhour%20cache%20lifetime%20applies.), so when we [create](https://github.com/flyteorg/flyteplugins/blob/f1feedeb49e2d1c103bffbbe0ad59293bc21162b/go/tasks/plugins/webapi/bigquery/plugin.go#L114) a bq task, the job status will return  “Done” instead of “running”. Because the status is done, propeller won’t invoke [get](https://github.com/flyteorg/flyteplugins/blob/46386ae52192e2984192d20d089c4a8f55103bab/go/tasks/plugins/webapi/bigquery/plugin.go#L189) which can help us construct [outputLocation](https://github.com/flyteorg/flyteplugins/blob/46386ae52192e2984192d20d089c4a8f55103bab/go/tasks/plugins/webapi/bigquery/plugin.go#L219) for structured dataset. Therefore, it causes the uri in structured dataset become empty

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
Add OutputLocation to ResourceWrapper if job status is Done

image

## Tracking Issue
_Remove the '*fixes*' keyword if there will be multiple PRs to fix the linked issue_

fixes https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_

### CODE REVIEW

File: go/tasks/plugins/webapi/bigquery/plugin.go [hunk 1]
```diff
@@ -291,6 +307,14 @@ func (p Plugin) Status(ctx context.Context, tCtx webapi.StatusContext) (phase co
 	return core.PhaseInfoUndefined, pluginErrors.Errorf(pluginsCore.SystemErrorCode, "unknown execution phase [%v].", resource.Status.State)
 }
 
+func constructOutputLocation(job *bigquery.Job) string {
+	dst := job.Configuration.Query.DestinationTable
+	if dst == nil {
+		return ""
+	}
+	return fmt.Sprintf("bq://%v:%v.%v", dst.ProjectId, dst.DatasetId, dst.TableId)
```
[the remark this answers is not in the document]
Comment (pingsutw) [reply go/tasks/plugins/webapi/bigquery/plugin.go:315]: we only use `bq://` in the bigquery task because BQ will write query results to a temporary table, and `bq://...` points to that table.

File: go/tasks/plugins/webapi/bigquery/plugin.go [hunk 2]
```diff
@@ -291,6 +307,14 @@ func (p Plugin) Status(ctx context.Context, tCtx webapi.StatusContext) (phase co
 	return core.PhaseInfoUndefined, pluginErrors.Errorf(pluginsCore.SystemErrorCode, "unknown execution phase [%v].", resource.Status.State)
 }
 
+func constructOutputLocation(job *bigquery.Job) string {
+	if job == nil || job.Configuration == nil || job.Configuration.Query == nil || job.Configuration.Query.DestinationTable == nil {
+		return ""
+	}
+	dst := job.Configuration.Query.DestinationTable
+	return fmt.Sprintf("bq://%v:%v.%v", dst.ProjectId, dst.DatasetId, dst.TableId)
```
Comment (wild-endeavor) [inline go/tasks/plugins/webapi/bigquery/plugin.go:315]: one more thing @pingsutw - can you debug log this?

Comment (pingsutw) [reply go/tasks/plugins/webapi/bigquery/plugin.go]: updated it

File: go/tasks/plugins/webapi/bigquery/plugin.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (honnix) [reply go/tasks/plugins/webapi/bigquery/plugin.go]: `gs://` refers to GCS and in this case it is about BigQuery.

**Review by wild-endeavor — APPROVED** [review]
